runtime/context.go

```diff
@@ -41,6 +41,13 @@ var (
 	DefaultContextTimeout = 0 * time.Second
 )
 
+// malformedHTTPHeaders are headers would be considered malformed and be rejected by grpc server
+// Connection: https://github.com/grpc/grpc-go/pull/4803
+var malformedHTTPHeaders = map[string]struct{}{
+	"Connection": {},
+	"connection": {},
```

Comment:
HTTP headers are not case sensitive, so we can just transform them to lower case when making the check.
```suggestion
	"connection": {},
```